### ISSUE TITLE
[Android] Accessibility for top bar buttons

### ISSUE THREAD

**Issue by feldmisco** [wix/react-native-navigation#904]
### Issue Description

When using TalkBack accessibility on Android, there isn't appropriate accessible information for back buttons or user added buttons.

Back button - Currently TalkBack will say "button, unlabeled" instead of something like "navigate up, button" for the back button. 

User-specified buttons - While it will currently announce the name of the buttons, it should also say the type of component afterwards. For example, with a search button, instead of "search", it should say "search, button".

### Steps to Reproduce / Code Snippets / Screenshots

Any screen that is pushed onto the navigation stack will have a back button. Enable TalkBack in Android, and tap the back button. It will announce "button, unlabeled".

---
### Environment
* React Native Navigation version: 2.0.0-experimental.233
* React Native version: 0.41.2
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Galaxy S7

**Comment by guyca** [comment]
@feldmisco Thanks for bringing this up to our attention. Do you have any experience with the TalkBack functionality on Android?

**Comment by feldmisco** [comment]
@guyca You're welcome. I have recent experience testing and implementing TalkBack functionality on Android from a React Native perspective, and Android development experience.

**Comment by GyyBest** [comment]
Any information for this feature, please? I'm using accessibility label for the navigation buttons, but no idea how to do this?

**Comment by GyyBest** [comment]
While I add title for the right buttons, TalkBack can read the accessibility label for the right buttons, but when I add backButtonTitle it doesn't work, while I use left buttons to add back buttons by add id: 'back', TalkBack still says 'Unlabeled button'

**Comment by GyyBest** [comment]
But it works well on the  ios phone

**Comment by bdebout** [comment]
Something new about that ? Or a workaround in order to label the back button ?

**Comment by gavin-brown** [comment]
I can also confirm that the unlabeled back button is happening on 1.1.336 with React Native 0.51.0.

**Comment by no23reason** [comment]
Are there any news on this issue? I can confirm it is still broken in `1.1.463`.

**Comment by ball-hayden** [comment]
Bumping this as not stale - TalkBack still reads "button, unlabelled" for back buttons.
There also doesn't appear to be a way of setting a label for an icon button (e.g. a hamburger button)
